bolt/test/X86/bug-reorder-bb-jrcxz.s

```diff
@@ -0,0 +1,643 @@
+# Test performs a BB reordering with unsupported 
+# instruction jrcxz. Reordering works correctly with the
+# follow options: None, Normal or Reverse. Other strategies 
+# are completed with Assertion `isIntN(Size * 8 + 1, Value).
+# The cause is the distance between BB where one contains 
+# jrcxz instruction.
+# Example: OpenSSL
+# https://github.com/openssl/openssl/blob/master/crypto/bn/asm/x86_64-mont5.pl#L3319
+
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -filetype=obj -triple x86_64-unknown-unknown \
+# RUN:   %s -o %t.o
+# RUN: link_fdata %s %t.o %t.fdata
+# RUN: %host_cc %t.o -falign-labels -march=native -o %t.exe -Wl,-q
+
+# RUN:  llvm-bolt %t.exe -o %t.bolted -data %t.fdata \
```

Comment:
Running BOLT on that input is crashing for me with:

/RewriteInstance.cpp:3266: void llvm::bolt::RewriteInstance::mapCodeSections(llvm::RuntimeDyld &): Assertion `TextSection->hasValidSectionID() && ".text section should be valid"' failed.

Run  "ninja check-bolt"  (or "make check-bolt") to see that, after configuring the project with -DLLVM_ENABLE_PROJECTS="bolt" (this is new for bolt in monorepo)

Reply:
done